src/components/burger-ingredients/options.tsx

```diff
@@ -2,11 +2,20 @@ import React from 'react';
 import styles from './burger-ingredients.module.css';
 import Ingredient from './ingredient';
 import Separator from './separator';
-import { data } from '../../utils/data.js'
-import { dataEl } from '../../utils/types.js'
+import IngredientDetails from '../ingredient-details/ingredient-details';
+import type { IngredientShape } from '../../utils/types.js'
 
-function Options() {
-  const orderFunc = (a:dataEl, b:dataEl) : number => {
+interface BurgerIngredientsProps{
+  data: IngredientShape[],
+  cart: IngredientShape[],
+  setCart: (newCart: IngredientShape[]|[]) => void,
+}
+
+const Options = (props: BurgerIngredientsProps) => {
+  const [detailsIngredient, setDetailsIngredient] = React.useState({} as IngredientShape);
```

Comment:
Можно лучше:
как дефолтное значение для `detailsIngredient` будет удобно засетать `null`, не нужно будет делать лишнее преобразование типа
+ нужно будет добавить возможную типизацию для хука:
```suggestion
  const [detailsIngredient, setDetailsIngredient] = React.useState<IngredientShape | null>(null);
```
и перед рендером `IngredientDetails` добавить проверку на существование этого ингредиента